app/controllers/users_controller.rb

```diff
@@ -0,0 +1,38 @@
+class UsersController < ApplicationController
+  before_action :check_authentication, only: :show
+  before_action :check_signup, only: [:new, :create]
+  def new
+    @user = User.new
+  end
+
+  def show
+    @user = current_user
+  end
+
+  def create
```

Comment:
Hi Ryan,
Ok, I made the change but I want you to confirm to me that the "All Posts" link works even if there isn't an account created because, on my end, I can see all the posts. Thank you!